UI/CardScrollUI.gd

```diff
@@ -0,0 +1,71 @@
+extends Control
+
+var card_scene: PackedScene = preload("res://Cards/Card.tscn")
+
+var discard_pile: Array[CardBase] = CardManager.card_container.discard_pile
+var draw_pile: Array[CardBase]    = CardManager.card_container.draw_pile
+var deck: Array[CardBase]         = CardManager.card_container.default_deck
+var cards_to_display: Array[CardBase] = []
+var card_worlds: Array[CardWorld] = []
+var cardworld: CardWorld = null
+
+func _on_button_pressed() -> void:
+	queue_free()
+
+#Pressing escape will return you to the game
+func _input(event) -> void:
+	if event.is_action("escape"):
+		queue_free()
+
+
+
+func populate(parent_name: String) -> void:
+	
+	match parent_name:
+		"DiscardPile":
+			cards_to_display = discard_pile
+			$Label.text = "Showing the discard pile"
+		"DrawPile":
+			cards_to_display = draw_pile
+			$Label.text = "Showing the draw pile"
+		"Deck":
+			cards_to_display = deck
+			$Label.text = "Showing the deck"
+		
+	
+	cards_to_display.sort()
+
+	for card: CardBase in cards_to_display:
+
+		cardworld = card_scene.instantiate()
+		$ScrollContainer/GridContainer.add_child(control_cardworld(cardworld))
+
+		cardworld.init_card(card)
+	
+	
+
+
+func control_cardworld(card: CardWorld) -> Control:
+
+	var control: Control = Control.new()
+	var cardui: Control = card.get_children()[2] # Gets the control node CardUI 
```

Comment:
Magic Number here, I would recommend to make a constant of the index 2 to specify exactly which node you're getting here. Something like:

```
const CONTROL_NODE_INDEX = 2
```

That way we can remove that comment you have right next to it.

Reply:
Oh right, Im going to that and make all of the arbitrary variables constant too for future stuff too